Project-Hotpot/Project-Hotpot/APIManager.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  APIManager.swift
+//  Project-Hotpot
+//
+//  Created by Harini Sundaram on 7/11/22.
+//
+
+import Foundation
+import UIKit
+
+class APIManager: NSObject {
+//    @IBOutlet weak var songTitle: UILabel!
+    var curr_song_label:String?
+    
+//    class NetworkManager {
+
+        // MARK: - Properties
+
+        private static var sharedAPIManager: APIManager = {
+            let SpotifyManager = APIManager()
+            return SpotifyManager
+        }()
+
+        // MARK: - Initializers
+        //doing this way so that only 1 instance can be created
+        override private init() {
+            NSLog("API Manager Initialized")
+        }
+
+        // MARK: - Accessors
+        
+        class func shared() -> APIManager {
+            return sharedAPIManager
+        }
+
+    
+    var lastPlayerState: SPTAppRemotePlayerState?
```

Comment:
Dont know if this is iOS standard or not (@toby1236 can better comment) but generally it is good to define all vars at the top of the class followed by functions. Within those publics go before privates. Additionally statics go to the top

Reply:
Ah yes, this is standard for iOS too. The more organized you keep mutable state, the easier it is to reason about bugs.